### PR TITLE
Closes #1619: Fix slow search page with custom metadata blocks

### LINKED ISSUE #1619: Dataverse search page is slow when we are using custom metadata block

### PR DESCRIPTION
**Pull request by wtr-icm**

### CODE REVIEW

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/common/BundleUtil.java [hunk 1]
```diff
@@ -123,23 +126,35 @@ private static String getStringFromInternalBundle(String bundleKey, String bundl
...
                     : StringUtils.EMPTY;
         } catch (Exception ex) {
-            logger.warning("Could not find key \"" + bundleKey + "\" in bundle file: " + bundleName);
+            logger.finest("Could not find key \"" + bundleKey + "\" in bundle file: " + bundleName);
             return StringUtils.EMPTY;
         }
     }
 
-    private static Optional<String> getStringFromExternalBundle(String bundleKey, String bundleName, Locale locale) {
+    // NB this method is nearly exact copy of getStringFromInternalBundle(…), however
```
Comment (dbojanek) [inline dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/common/BundleUtil.java]: what NB means here? Maybe consider making it a JavaDoc for readability reasons?

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/common/BundleUtil.java [hunk 2]
```diff
@@ -123,23 +126,35 @@ private static String getStringFromInternalBundle(String bundleKey, String bundl
...
+    // any trial to extract common code from these two and pass differing parts as lambdas
```
Comment (dbojanek) [inline dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/common/BundleUtil.java]:
```suggestion
    // any attempt in extracting common code from these two and pass differing parts as lambdas
```

**Review by dbojanek — APPROVED** [review]
Are you able to explain why was that causing performance issues? How did you find this one out?

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/common/BundleUtil.java (hunk 1 above)
Comment (wtr-icm) [reply dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/common/BundleUtil.java]: NB = Nota Bene, however I've changed to IMPORTANT.
I'd rather not change that comment to javadoc, as it's only about code details.

**Pull request merged**